packages/beacon-node/src/network/metadata.ts

```diff
@@ -35,7 +34,7 @@ export interface IMetadataModules {
  * https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/phase0/p2p-interface.md#metadata
  */
 export class MetadataController {
-  private enr?: ENR;
+  private setEnrValue?: (key: string, value: Uint8Array) => Promise<void>;
```

Comment:
Someone has to 'own' the ENR.
Current choice is for the discv5 worker to own the ENR and just pass a function to update it inside the worker.

The FileENR should also be removed and we just write the ENR to disk on close.